backend/mora/auth/base.py

```diff
@@ -42,11 +42,16 @@ def get_user():
     :return: The username of the user who is currently logged in.
     '''
 
-    username_attr = flask.current_app.config['SAML_USERNAME_ATTR']
-    try:
-        username = flask_saml_sso.get_session_attributes().get(
-            username_attr)[0]
-    except (AttributeError, IndexError):
-        username = None
+    if not flask.current_app.config['SAML_USERNAME_FROM_NAMEID']:
+        username_attr = flask.current_app.config['SAML_USERNAME_ATTR']
+        try:
+            username = flask_saml_sso.get_session_attributes().get(
+                username_attr)[0]
+        except (AttributeError, IndexError, TypeError):
+            flask.current_app.logger.exception(
```

Comment:
Perhaps?

```suggestion
            util.log_exception(
```

(Although it might cause issues with confidentiality depending on the actual request.)